Submitted By:            DJ Lucas
Date:                    2015-12-18
Initial Package Version: 228
Upstream Status:         Applied
Origin:                  https://github.com/systemd/systemd/commit/3ccd31635353
Description:             Some routers (dnsmasq) will send packets from global
                         addresses, which would break the default route setup,
                         so ignore those.

--- systemd-228-orig/src/libsystemd-network/sd-ndisc.c	2015-12-18 00:38:09.973606527 -0600
+++ systemd/src/libsystemd-network/sd-ndisc.c	2015-12-18 01:03:10.829172474 -0600
@@ -561,6 +561,9 @@ static int ndisc_router_advertisment_rec
                 return 0;
         }
 
+        if (!in_addr_is_link_local(AF_INET6, (const union in_addr_union*) &router.in6.sin6_addr))
+                return 0;
+
         if (ra->nd_ra_type != ND_ROUTER_ADVERT)
                 return 0;
 
